File: backend/domain/common/interfaces/entity.js
/**
 * 实体基类
 * 所有领域实体都应继承此类
 */
class Entity {
  /**
   * @param {string} id - 实体的唯一标识符
   */
  constructor(id) {
    if (!id) {
      throw new Error('实体必须有ID');
    }
    this._id = id;
  }

  /**
   * 获取实体ID
   * @returns {string} 实体ID
   */
  get id() {
    return this._id;
  }

  /**
   * 比较两个实体是否相等
   * 实体相等基于ID相等，而非属性相等
   * @param {Entity} entity - 要比较的实体
   * @returns {boolean} 是否相等
   */
  equals(entity) {
    if (!(entity instanceof Entity)) {
      return false;
    }
    return this._id === entity.id;
  }
}

module.exports = Entity; 

File: frontend/flutter/lib/domain/common/entities/entity.dart
/// 实体基类
/// 
/// 所有领域实体都应该继承此类
/// 提供实体的基本功能和约束
abstract class Entity {
  const Entity();
  
  /// 判断两个实体是否相等
  /// 基于运行时类型和属性进行比较
  bool equals(Entity other) {
    return identical(this, other);
  }
}